File: editor/src/js/http/AuthApi.ts
import {HttpService} from "../common/services/HttpService";
import {LoginRequest} from "../model/LoginRequest";
import {User} from "../model/User";
import {CommonError} from "../model/common/CommonError";

export class AuthApi {
  constructor(private http: HttpService) {
  }

  userGetGet(): Promise<User> {
    return this.http.GET(`/user.get`);
  }
  userLoginPost(loginRequest: LoginRequest): Promise<any> {
    return this.http.POST(`/user.login`, loginRequest);
  }
  userLogoutPost(): Promise<any> {
    return this.http.POST(`/user.logout`);
  }
}

File: editor-api/src/main/groovy/ish/oncourse/willow/editor/service/AuthApi.java
package ish.oncourse.willow.editor.service;

import ish.oncourse.willow.editor.model.LoginRequest;
import ish.oncourse.willow.editor.model.User;
import ish.oncourse.willow.editor.model.common.CommonError;

import java.io.InputStream;
import java.io.OutputStream;
import java.util.List;
import java.util.Map;
import javax.ws.rs.*;
import javax.ws.rs.core.Response;
import javax.ws.rs.core.MediaType;

@Path("/")
public interface AuthApi  {

    @GET
    @Path("/user.get")
    @Consumes({ "application/json" })
    @Produces({ "application/json" })
    @AuthFilter
    User userGetGet();

    @POST
    @Path("/user.login")
    @Consumes({ "application/json" })
    @Produces({ "application/json" })
    void userLoginPost(LoginRequest loginRequest);

    @POST
    @Path("/user.logout")
    @Consumes({ "application/json" })
    @Produces({ "application/json" })
    void userLogoutPost();
}

